### PR TITLE
Core/auto update version env var

### PR DESCRIPTION
**Pull request by clairesunstudio**
This PR is to add the update core version task into the auto release script, so that the mayflower CDN version in the core documentation is updated during the release process.

To test:
`npm install -g js-yaml@3.13.1 @octokit/rest semver@6.3.0 shelljs@0.8.3 simple-git@1.126.0`
`export NODE_PATH=$(npm root --quiet -g) && node scripts/release-branch.js`

### CODE REVIEW

**Comment by clairesunstudio** [comment]
@smurrayatwork  I can't figure out why the `version` passed into updateCoreVersion(version) is undefined. 

In `.env` after running the release-branch script:

```
VERSION=@undefined
```

File: scripts/update-version.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const path = require('path');
+const fs = require('fs');
+
+module.exports = function updateCoreVersion(newVer) {
```
Comment (smurrayatwork) [inline scripts/update-version.js:4]: This function should be made to be async and awaited in `scripts/release-branch.js`:
```suggestion
module.exports = async function updateCoreVersion(newVer) {
```

File: scripts/release-branch.js [hunk 2]
```diff
@@ -18,17 +19,17 @@ const releaseBranch = 'release/' + version;
   // This asynchronous logic will happen sequentially.
   // If an error is thrown, it will break out of this
   // asynchronous function immediately and exit 1.
-
   // Create the release branch and push to Github.
   shell.exec(`git branch -D ${releaseBranch}`)
+  updateCoreVersion(version)
```
Comment (smurrayatwork) [inline scripts/release-branch.js:24]: Once updateCoreVersion is updated to be async:
```suggestion
  await updateCoreVersion(version)
```

File: scripts/update-version.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  const filePath = `${path.resolve(__dirname, '../packages/core')}/.env`;
+
+  const content = fs.readFileSync(filePath).toString();
```
Comment (smurrayatwork) [inline scripts/update-version.js:7]:
```suggestion
  try {
    const content = await fs.readFile(filePath);
```

File: scripts/update-version.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const path = require('path');
+const fs = require('fs');
```
Comment (smurrayatwork) [inline scripts/update-version.js:2]:
```suggestion
const fs = require('fs').promises;
```

File: scripts/update-version.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+module.exports = function updateCoreVersion(newVer) {
+  const filePath = `${path.resolve(__dirname, '../packages/core')}/.env`;
+
+  const content = fs.readFileSync(filePath).toString();
+
+
+  const regex = /[0-9]+\.[0-9]+\.[0-9]+/g;
+  const newContent = content.replace(regex, newVer)
+
+  fs.writeFileSync(filePath, newContent, (err) => {
```
Comment (smurrayatwork) [inline scripts/update-version.js:13]:
```suggestion
  await fs.writeFile(filePath, newContent);
```

File: scripts/update-version.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    if (err) throw err;
```
Comment (smurrayatwork) [inline scripts/update-version.js:14]:
```suggestion
    } catch(error) {
      throw error;
    }
```

File: scripts/update-version.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  })
```
Comment (smurrayatwork) [inline scripts/update-version.js:15]:
```suggestion
```

File: scripts/update-version.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const path = require('path');
+const fs = require('fs').promises;
+
+module.exports = async function updateCoreVersion(newVer) {
+  const filePath = `${path.resolve(__dirname, '../packages/core')}/.env`;
+
+  try {
+    const content = fs.readFileSync(filePath).toString();
```
Comment (smurrayatwork) [inline scripts/update-version.js]: Shouldn't this be async too?
```suggestion
    const content = await fs.readFile(filePath).toString()

**Review by smurrayatwork — APPROVED** [review]

**Pull request merged**